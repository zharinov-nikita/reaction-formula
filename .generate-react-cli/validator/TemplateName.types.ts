export interface ValidatorTemplateNameProps {}
export interface ValidatorTemplateName {}
