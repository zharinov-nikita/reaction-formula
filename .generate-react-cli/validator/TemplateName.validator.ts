import { ValidatorTemplateName, ValidatorTemplateNameProps } from './template-name.types'

export function validatorTemplateName(props: ValidatorTemplateNameProps): ValidatorTemplateName {
  return {}
}
