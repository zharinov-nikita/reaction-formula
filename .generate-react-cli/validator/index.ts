export * from './template-name.validator'
export * from './template-name.types'