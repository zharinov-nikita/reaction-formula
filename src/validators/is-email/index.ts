export * from './is-email.validator'
export * from './is-email.types'
