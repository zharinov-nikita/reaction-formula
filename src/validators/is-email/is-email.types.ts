export type ValidatorIsEmailProps = string 
export type ValidatorIsEmail = boolean
