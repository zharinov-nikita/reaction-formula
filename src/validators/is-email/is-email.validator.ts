import { ValidatorIsEmail, ValidatorIsEmailProps } from './is-email.types'

export function validatorIsEmail(value: ValidatorIsEmailProps): ValidatorIsEmail {
  const regexp =
    /^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i
  return regexp.test(value)
}
