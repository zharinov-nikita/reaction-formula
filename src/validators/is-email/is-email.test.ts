import { validatorIsEmail } from './is-email.validator'

describe('validatorIsEmail', () => {
  test('expected true', () => {
    expect(validatorIsEmail('user@gmail.com')).toBe(true)
  })

  test('expected false', () => {
    expect(validatorIsEmail('user@gmail')).toBe(false)
  })

  test('snapshot', () => {
    expect(validatorIsEmail('user@gmail.com')).toMatchSnapshot()
    expect(validatorIsEmail('user@gmail')).toMatchSnapshot()
  })
})
