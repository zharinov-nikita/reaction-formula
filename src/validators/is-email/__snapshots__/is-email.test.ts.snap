// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validatorIsEmail snapshot 1`] = `true`;

exports[`validatorIsEmail snapshot 2`] = `false`;

exports[`validatorIsEmail snapshot: snapshot 1`] = `[Function]`;
